src/util/jira-utils.ts

```diff
@@ -105,3 +107,13 @@ export const jiraIssueKeyParser = (str: string): string[] => {
 };
 
 export const hasJiraIssueKey = (str:string): boolean => !isEmpty(jiraIssueKeyParser(str));
+
+export const isGitHubCloudApp = async (jiraHost: string): Promise<Promise<boolean> | boolean>=> {
+	if (await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)) {
+		const installation = await Installation.getForHost(jiraHost);
+		const gitHubAppId = installation?.githubAppId;
+		return gitHubAppId ? !await GitHubServerApp.getForGitHubServerAppId(gitHubAppId) : true;
+	}
```

Comment:
Got in a tangle reading this one.
Is this right?
if GHE FF is off then cloud?
if no gitHubAppId then cloud?
if githubcloudId but doesn't find anything then cloud?

my opinion piece is I would break it up a bit just for readability.

Reply:
Yeah basically, we want to return true if cloud and false if GHE, and we detect a GHE if there is an entry in the GitHubServerApps table.

Any suggestions on how you want to break it out?

Reply:
untested.. but something like this?  ¯\_(ツ)_/¯
Can rip out the feature flag easy when the time comes too.
```
export const isGitHubCloudApp = async (jiraHost: string): Promise<boolean>=> {
	if (await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)) {
		return false;
	}

	const installation = await Installation.getForHost(jiraHost);
	const gitHubAppId = installation?.githubAppId;
////githubcomment - don't need the **gitHubAppId &&** since getForGitHubServerAppId will return null, but easier to understand and can stop unnecessary calls......
/// i'm also assuming that this.findOne in getForGitHubServerAppId returns null or undefined if it doesn't find anything
if(gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId)) {
		return false;
	}
	return true;
}
```

Reply:
![Screen Shot 2022-05-31 at 9 34 20 am](https://user-images.githubusercontent.com/37155488/171068174-c20bb639-688a-41a2-ae27-51e70188e21c.png)

IntelliJ is complaining that we can simplify the last conditional. It suggests changing it to `return !(gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId));` Thoughts? Listen to IntelliJ or ignore it?

Reply:
I'd suggest to have some temp var like `isGitHubEnterpriseServer=gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId)`, and then `return !isGitHubEnterpriseServer`...

Reply:
Updated. Lmk what you think. I'd added some comments for clarity (which is not ideal but we'll be able to clean this function up when we remove the FF). I had to modify the suggestions a little. Couldn't have:
```
if (await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost)) { return false; }
```
because it would return false in instances where the feature flag is turned on but there is no gitHubAppId or entry in the GitHubServerApps table (github cloud app).

Reply:
a lot easier to understand I reckon.
Yea once that FF goes you could probably do what intelliJ said and it will be clean as!

Reply:
do you even need the FF in there?  githubAppId will be null if it's cloud.  You could literally just do `return !(await Installation.getForHost(jiraHost)).githubAppId`